### PR TITLE
Fix Story Duration Default

### LINKED ISSUE #12599: Story Duration Default - No notification to user who tries to set it higher than 20 seconds
## Bug Description

Originally reported by @barklund 
 If you set the default interval to a number higher than 20, it’s capped at 20, but with no indication that such a cap exists. Same thing happens inside the editor. We should have some sort of notification & description

## Expected Behaviour

The user should be notified that there is a max value, and that going over that value isn't possible

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### LINKED ISSUE #12600: Story Duration Default - toggling the manual/auto switch reverts the duration to it's previous value
## Bug Description

Originally reported by @timarney 

Setup x seconds under the dashboard… create a story look at the document settings page advancement …. This has the current value.  Still under the story swap to manual and back to auto — the time now shows 7 secs vs the dashboard settings value.

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### LINKED ISSUE #12644: Story Duration Default - manual setting is not persisted
## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by merapi**
## Summary

This PR fixes a few issues listed at the bottom.

For https://github.com/googleforcreators/web-stories-wp/issues/12599 I tried to use snackbar, but because the min/max logic is inside the input component that might be too invasive.
image
image

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12600
Fixes #12599
Fixes #12644

### CODE REVIEW

**Comment by kkalarickal** [comment]
@merapi - I checked out this PR per your go ahead, using the `12645 branch` this evening. I can confirm that the global page advancement setting is maintained and if set to auto-advance, the duration can only be between 1 and 20. The global setting is used when toggling between manual and auto-adv, and when user creates a story from template also.

A few observations that might be worth confirming:
1. decimals are no longer possible
2. duration numbers with decimals more than 1 sec are ALWAYS rounded down - so 10..99 becomes 10, instead of 11.

Can you confirm this behavior is ok ?

**Comment by merapi** [comment]
Thanks for testing. I think I rushed with the decimals part, we will discuss it further.

**Comment by merapi** [comment]
> A few observations that might be worth confirming:
> 
> 1. decimals are no longer possible
> 2. duration numbers with decimals more than 1 sec are ALWAYS rounded down - so 10..99 becomes 10, instead of 11.
> 
> Can you confirm this behavior is ok ?

I'm ok with that, it is consistent with other non-decimal fields. Pascal is also ok: https://github.com/GoogleForCreators/web-stories-wp/issues/12599#issuecomment-1306975497

File: packages/story-editor/src/components/panels/shared/generalPageAdvancement.js [hunk 1]
```diff
@@ -164,20 +164,29 @@ function GeneralPageAdvancementPanel({
...
+              disabled={customAdvancementDisabled}
+              containerStyleOverride={inputContainerStyleOverride}
+            />
+          </Row>
+          <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+            {sprintf(
+              /* translators: 1: minimum duration. 2: maximum duration. */
+              __('Duration between %1$d and %2$d seconds.', 'web-stories'),
+              MIN_MAX.PAGE_DURATION.MIN,
+              MIN_MAX.PAGE_DURATION.MAX
```
Comment (swissspidy) [inline packages/story-editor/src/components/panels/shared/generalPageAdvancement.js]: this should use `_n()` to cater for pluralization in different languages

Comment (merapi) [reply packages/story-editor/src/components/panels/shared/generalPageAdvancement.js]: @swissspidy https://github.com/GoogleForCreators/web-stories-wp/pull/12645/commits/24d5dc1bc7b22058399b7e79d42ecd7a11f67634

**Review by swissspidy — APPROVED** [review]

**Pull request merged**